src/treeview/DbConnectionUpdater.ts

```diff
@@ -1,27 +1,60 @@
 import * as vscode from "vscode";
 import { DatabaseProcessor } from "../db/DatabaseProcessor";
-import { IConfig } from "../view/app/model";
+import { ConnectionStatus, IConfig } from "../view/app/model";
 import { IRefreshCallback, RefreshWithoutCallback } from "./IRefreshCallback";
 
 export class DbConnectionUpdater {
     constructor (){}
+    private locked: boolean = false;
 
     public async updateConnectionStatuses (context: vscode.ExtensionContext) {
         this.updateConnectionStatusesWithRefreshCallback(context, new RefreshWithoutCallback());
     }
 
     public async updateConnectionStatusesWithRefreshCallback (context: vscode.ExtensionContext, refreshCallback: IRefreshCallback) {
+        if (this.locked === false){
+            this.locked = true;
+            
+            await this.updateStatuses(context,refreshCallback);
+
+            this.locked = false;
+        }
+    }
+
+    private async updateStatuses(context: vscode.ExtensionContext, refreshCallback: IRefreshCallback){

```

Comment:
Probably its better to move context to a class variable and set it in `updateConnectionStatusesWithRefreshCallback`